conceptnet5/db/prepare_data.py

```diff
@@ -28,24 +28,54 @@ def sanitize(text):
     return text.replace('\n', '').replace('\t', '').replace('\\', '\\\\')
 
 
+def gin_indexable_edge(edge):
+    """
+    Convert an edge into a dictionary that can be matched with the JSONB @>
+    operator, which tests if one dictionary includes all the information in
+    another. This operator can be indexed by GIN.
+
+    We replace the 'start', 'end', 'rel', and 'dataset' URIs with lists
+    of their URI prefixes. We query those slots with a single-element list,
+    which will be a sub-list of the prefix list if it's a match.
+
+    As an example, a query for {'start': '/c/en'} will become the GIN
+    query {'start': ['/c/en']}, which will match indexed edges such as
+    {
+        'start': ['/c/en', '/c/en/dog'],
+        'end': ['/c/en', '/c/en/bark'],
+        'rel': ['/r/CapableOf'],
+        ...
+    }
+    """
+    gin_edge = {}
+    gin_edge['uri'] = edge['uri']
+    gin_edge['start'] = uri_prefixes(edge['start'])
+    gin_edge['end'] = uri_prefixes(edge['end'])
+    gin_edge['rel'] = uri_prefixes(edge['rel'])
```

Comment:
I think `uri_prefixes` will always just return the relation itself, for example `/r/CapableOf`, so it should be enough to say `gin_edge['rel'] = edge['rel']`.

Reply:
It still needs to be a list, and we might as well be consistent. This additionally allows matching all the DBPedia relations that are "on probation" in their own namespace, like `/r/dbpedia/genre`, by querying for `/r/dbpedia`.